analysis/deep2_r23_o32/plotting/te_metal_plots.py

```diff
@@ -79,37 +79,47 @@ def plotting_te_metal(fitspath, fitspath_ini, revised=False):
             -> refers to if using the bin_derived_prop_revised temperature
                      and metallicity measurements which right now implement dust attenuation
     """
-    indv_all_file = join(fitspath, filename_dict['indv_bin_info'])
+    if raw:
+        mc = '.MC'
+    else:
+        mc = ''
 
+    if apply_dust:
+        ad = '.dustcorr'
+    else:
+        ad = ''
     if revised:
-        composite_file = join(fitspath, filename_dict['bin_derived_prop_rev'])
-        out_pdf = join(fitspath, 'temperature_metallicity_revised.pdf')
+        rev_s = ''
+        verification = asc.read(join(fitspath, filename_dict['bin_valid_rev']))
     else:
-        composite_file = join(fitspath, filename_dict['bin_derived_prop'])
-        out_pdf = join(fitspath, 'temperature_metallicity_plots.pdf')
+        rev_s = '.valid1'
+        verification = asc.read(join(fitspath, filename_dict['bin_valid']))
+    out_pdf = fitspath + 'temperature_metallicity_plots' + rev_s + mc + ad + '.pdf'
+    tempature_table = fitspath + 'bin_derived_properties' + rev_s + mc + ad + '.tbl'
+    print(tempature_table)
+    comp_derived = asc.read(tempature_table)
 
-    # Composite bin derived properties
-    comp_derived = asc.read(composite_file)
     # Individual Measurements
-    indv_all = asc.read(indv_all_file)
-
-    icom_log = indv_all['12+log(O/H)']
-    ilogR23 = indv_all['logR23']
-    ilogO32 = indv_all['logO32']
-    print(ilogR23)
+    if individual:
+        indv_all_file = join(fitspath, filename_dict['indv_bin_info'])
+        indv_all = asc.read(indv_all_file)
 
-    # icom_nan = np.isnan(icom_log)
-    iidx = np.where((icom_log != 0.0))[0]
+        icom_log = indv_all['12+log(O/H)']
+        ilogR23 = indv_all['logR23']
+        ilogO32 = indv_all['logO32']
+        iidx = np.where((icom_log != 0.0))[0]
 
-    iiR23_idv = ilogR23[iidx]
-    iiO32_idv = ilogO32[iidx]
-    iR23_idv = np.log10(iiR23_idv)
-    iO32_idv = np.log10(iiO32_idv)
-    print("len:", len(iR23_idv))
+        iiR23_idv = ilogR23[iidx]
+        iiO32_idv = ilogO32[iidx]
+        iR23_idv = np.log10(iiR23_idv)
+        iO32_idv = np.log10(iiO32_idv)
+        print("len:", len(iR23_idv))
 
     # DEEP2 and MACT Data
-    derived = asc.read(join(fitspath_ini, 'DEEP2_R23_O32_derived.tbl'))
-    derived_MACT = asc.read(join(fitspath_ini, 'MACT_R23_O32_derived.tbl'))
+    derived_table = fitspath_ini + '/DEEP2_Commons/Catalogs/DEEP2_R23_O32_derived.tbl'
+    MACT_table = fitspath_ini + '/MACT_Commons/Catalogs/MACT_R23_O32_derived.tbl'
```

Comment:
Please use `join` here. 

These variable names are confusing. You have `derived_table` referring to a file but `derived` referring to the table.

Reply:
Used join and changed variable names